src/home/getters/get-github-user.ts

```diff
@@ -30,13 +27,38 @@ async function getNewSessionToken(): Promise<string | null> {
   const params = new URLSearchParams(hash.substr(1)); // remove the '#' and parse
   const providerToken = params.get("provider_token");
   if (!providerToken) {
-    return null;
+    const err = params.get("error");
+    const code = params.get("error_code");
+    const desc = params.get("error_description");
+
+    if (err === "server_error") {
+      if (code === "500") {
+        if (desc === "Error getting user profile from external provider") {
```

Comment:
Is there no better check that this? If one letter in the text changes it breaks the whole login. Also `500` and `server_error` seem pretty redundant.

Reply:
I don't think so no but I've removed the err and code checks

It's the Supabase response so not much control over it, the test-case I found it in was that I had maxed out PAT

Reply:
I mean that the only check here should be `code === '500'`. `err` is just the translation of the `code` to readable text, and `desc` just sent by Supabase as a default error message.

You can use it to output the error instead:
`console.error(desc);`

Reply:
Okay mate I understand, I was trying to be specific to the edge-case that way I wasn't handling other failures with the token expired response. Because ``code == 500`` could represent multiple errors, I remember mentioning it previously re: generic vs specific approach to handling it.

But what'll do is just `console.err` the desc if `code == 500` and fire the toaster regardless

Reply:
checking only that `code === '500' and if so console.err and show toaster to user